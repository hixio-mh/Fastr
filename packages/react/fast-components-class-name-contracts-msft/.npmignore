# Source files
src/

# Configs
tsconfig.json