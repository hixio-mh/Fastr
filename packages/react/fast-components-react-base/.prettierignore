coverage/*
dist/*
www/*
src/**/.tmp/