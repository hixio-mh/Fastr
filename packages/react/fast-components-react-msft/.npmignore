# Tests
__test__/
*.spec.*
*.test.*

# Source files
app/
www/
build/
coverage/
src/
assets/

# config
babel.config.js
tsconfig.json
webpack.config.js