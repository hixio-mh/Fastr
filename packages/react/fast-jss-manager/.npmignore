# Source files
src/

# config
tsconfig.json