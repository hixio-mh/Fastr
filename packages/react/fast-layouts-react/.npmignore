# Tests
__test__/
*.spec.*
*.test.*

# Source files
app/
coverage/
src/
www/

# config
babel.config.js
tsconfig.json
webpack.config.js