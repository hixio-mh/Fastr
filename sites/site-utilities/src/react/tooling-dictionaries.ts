import {
    AccentButton,
    accentButtonSchema2,
    ActionToggle,
    actionToggleSchema2,
    ActionTrigger,
    actionTriggerSchema2,
    AutoSuggest,
    AutoSuggestOption,
    autoSuggestOptionSchema2,
    autoSuggestSchema2,
    Badge,
    badgeSchema2,
    Breadcrumb,
    breadcrumbSchema2,
    Button,
    buttonSchema2,
    CallToAction,
    callToActionSchema2,
    Caption,
    captionSchema2,
    Card,
    cardSchema2,
    Carousel,
    carouselSchema2,
    Checkbox,
    checkboxSchema2,
    ContextMenu,
    ContextMenuItem,
    contextMenuItemSchema2,
    contextMenuSchema2,
    Dialog,
    dialogSchema2,
    Divider,
    dividerSchema,
    Flipper,
    flipperSchema2,
    Heading,
    headingSchema2,
    Hypertext,
    hypertextSchema2,
    Image,
    imageSchema,
    Label,
    labelSchema2,
    LightweightButton,
    lightweightButtonSchema2,
    Metatext,
    metatextSchema2,
    NeutralButton,
    neutralButtonSchema2,
    NumberField,
    numberFieldSchema,
    OutlineButton,
    outlineButtonSchema2,
    Paragraph,
    paragraphSchema2,
    Pivot,
    pivotSchema2,
    Progress,
    progressSchema2,
    Radio,
    radioSchema2,
    Select,
    SelectOption,
    selectOptionSchema2,
    selectSchema2,
    Slider,
    SliderLabel,
    sliderLabelSchema2,
    sliderSchema2,
    StealthButton,
    stealthButtonSchema2,
    Subheading,
    subheadingSchema2,
    TextAction,
    textActionSchema2,
    TextArea,
    textAreaSchema,
    TextField,
    textFieldSchema,
    Toggle,
    toggleSchema2,
    TreeView,
    TreeViewItem,
    treeViewItemSchema2,
    treeViewSchema2,
    Typography,
    typographySchema2,
} from "@microsoft/fast-components-react-msft";
import { ComponentDictionary } from "@microsoft/fast-tooling-react";
import { SchemaDictionary } from "@microsoft/fast-tooling";

export const reactComponentSchemaDictionary: SchemaDictionary = {
    [accentButtonSchema2.id]: accentButtonSchema2,
    [actionToggleSchema2.id]: actionToggleSchema2,
    [actionTriggerSchema2.id]: actionTriggerSchema2,
    [autoSuggestOptionSchema2.id]: autoSuggestOptionSchema2,
    [autoSuggestSchema2.id]: autoSuggestSchema2,
    [badgeSchema2.id]: badgeSchema2,
    [breadcrumbSchema2.id]: breadcrumbSchema2,
    [buttonSchema2.id]: buttonSchema2,
    [callToActionSchema2.id]: callToActionSchema2,
    [captionSchema2.id]: captionSchema2,
    [cardSchema2.id]: cardSchema2,
    [carouselSchema2.id]: carouselSchema2,
    [checkboxSchema2.id]: checkboxSchema2,
    [contextMenuItemSchema2.id]: contextMenuItemSchema2,
    [contextMenuSchema2.id]: contextMenuSchema2,
    [dialogSchema2.id]: dialogSchema2,
    [dividerSchema.id]: dividerSchema,
    [flipperSchema2.id]: flipperSchema2,
    [headingSchema2.id]: headingSchema2,
    [hypertextSchema2.id]: hypertextSchema2,
    [imageSchema.id]: imageSchema,
    [labelSchema2.id]: labelSchema2,
    [lightweightButtonSchema2.id]: lightweightButtonSchema2,
    [metatextSchema2.id]: metatextSchema2,
    [neutralButtonSchema2.id]: neutralButtonSchema2,
    [outlineButtonSchema2.id]: outlineButtonSchema2,
    [paragraphSchema2.id]: paragraphSchema2,
    [pivotSchema2.id]: pivotSchema2,
    [progressSchema2.id]: progressSchema2,
    [radioSchema2.id]: radioSchema2,
    [selectOptionSchema2.id]: selectOptionSchema2,
    [selectSchema2.id]: selectSchema2,
    [sliderLabelSchema2.id]: sliderLabelSchema2,
    [sliderSchema2.id]: sliderSchema2,
    [stealthButtonSchema2.id]: stealthButtonSchema2,
    [subheadingSchema2.id]: subheadingSchema2,
    [textActionSchema2.id]: textActionSchema2,
    [textAreaSchema.id]: textAreaSchema,
    [textFieldSchema.id]: textFieldSchema,
    [toggleSchema2.id]: toggleSchema2,
    [treeViewItemSchema2.id]: treeViewItemSchema2,
    [treeViewSchema2.id]: treeViewSchema2,
    [typographySchema2.id]: typographySchema2,
};

export const reactComponentDictionary: ComponentDictionary = {};
reactComponentDictionary[accentButtonSchema2.id] = AccentButton;
reactComponentDictionary[actionToggleSchema2.id] = ActionToggle;
reactComponentDictionary[actionTriggerSchema2.id] = ActionTrigger;
reactComponentDictionary[autoSuggestOptionSchema2.id] = AutoSuggestOption;
reactComponentDictionary[autoSuggestSchema2.id] = AutoSuggest;
reactComponentDictionary[badgeSchema2.id] = Badge;
reactComponentDictionary[breadcrumbSchema2.id] = Breadcrumb;
reactComponentDictionary[buttonSchema2.id] = Button;
reactComponentDictionary[callToActionSchema2.id] = CallToAction;
reactComponentDictionary[captionSchema2.id] = Caption;
reactComponentDictionary[cardSchema2.id] = Card;
reactComponentDictionary[carouselSchema2.id] = Carousel;
reactComponentDictionary[checkboxSchema2.id] = Checkbox;
reactComponentDictionary[contextMenuItemSchema2.id] = ContextMenuItem;
reactComponentDictionary[contextMenuSchema2.id] = ContextMenu;
reactComponentDictionary[dialogSchema2.id] = Dialog;
reactComponentDictionary[dividerSchema.id] = Divider;
reactComponentDictionary[flipperSchema2.id] = Flipper;
reactComponentDictionary[headingSchema2.id] = Heading;
reactComponentDictionary[hypertextSchema2.id] = Hypertext;
reactComponentDictionary[imageSchema.id] = Image;
reactComponentDictionary[labelSchema2.id] = Label;
reactComponentDictionary[lightweightButtonSchema2.id] = LightweightButton;
reactComponentDictionary[metatextSchema2.id] = Metatext;
reactComponentDictionary[neutralButtonSchema2.id] = NeutralButton;
reactComponentDictionary[numberFieldSchema.id] = NumberField;
reactComponentDictionary[outlineButtonSchema2.id] = OutlineButton;
reactComponentDictionary[paragraphSchema2.id] = Paragraph;
reactComponentDictionary[pivotSchema2.id] = Pivot;
reactComponentDictionary[progressSchema2.id] = Progress;
reactComponentDictionary[radioSchema2.id] = Radio;
reactComponentDictionary[selectOptionSchema2.id] = SelectOption;
reactComponentDictionary[selectSchema2.id] = Select;
reactComponentDictionary[sliderLabelSchema2.id] = SliderLabel;
reactComponentDictionary[sliderSchema2.id] = Slider;
reactComponentDictionary[stealthButtonSchema2.id] = StealthButton;
reactComponentDictionary[subheadingSchema2.id] = Subheading;
reactComponentDictionary[textActionSchema2.id] = TextAction;
reactComponentDictionary[textAreaSchema.id] = TextArea;
reactComponentDictionary[textFieldSchema.id] = TextField;
reactComponentDictionary[toggleSchema2.id] = Toggle;
reactComponentDictionary[treeViewItemSchema2.id] = TreeViewItem;
reactComponentDictionary[treeViewSchema2.id] = TreeView;
reactComponentDictionary[typographySchema2.id] = Typography;
