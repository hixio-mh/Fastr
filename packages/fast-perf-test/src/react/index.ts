import React from "react";
import ReactDOM from "react-dom";
import { Checkbox, TextField } from "@microsoft/fast-components-react-msft";

const root = document.createElement("div");
document.body.appendChild(root);

function getQueryVariable(variable) {
    var query = window.location.search.substring(1);
    var vars = query.split("&");
    for (var i = 0; i < vars.length; i++) {
        var pair = vars[i].split("=");
        if (decodeURIComponent(pair[0]) == variable) {
            return decodeURIComponent(pair[1]);
        }
    }
}

const count = getQueryVariable("count");

function renderItems() {
    return React.createElement(
        React.Fragment,
        {},
        new Array(typeof count === "string" ? (count as any) * 1 : 2000)
            .fill(0)
            .map((item, index) => React.createElement(TextField))
    );
}

function render(): void {
    ReactDOM.render(React.createElement(renderItems), root);
}

render();
