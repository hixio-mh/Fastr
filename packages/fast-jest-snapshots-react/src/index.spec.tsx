import * as React from "react";
import { generateSnapshots, SnapshotTestSuite } from "./index";

interface TestComponentProps {
    name: string;
}

class TestComponent extends React.Component<TestComponentProps, {}> {
    public render(): React.ReactNode {
        return <div>{this.props.name}</div>;
    }
}

const testSuite: SnapshotTestSuite<TestComponentProps> = {
    name: "TestComponent",
    component: TestComponent,
    data: [
        { name: "hello world" },
        [
            "should render with a snapshot title",
            {
                name: "hello pluto",
            },
        ],
    ],
};

/*
 * Test snapshot generation
 */
generateSnapshots(testSuite);

test("should not throw if no data is provided", (): void => {
    const adjustedTestSuite: any = Object.assign({}, testSuite);
    delete adjustedTestSuite.data;

    expect(
        (): void => {
            generateSnapshots(adjustedTestSuite);
        }
    ).not.toThrow();
});
