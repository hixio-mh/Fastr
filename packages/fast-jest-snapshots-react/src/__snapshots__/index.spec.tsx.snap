// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent should render with a snapshot title 1`] = `
<div>
  hello pluto
</div>
`;

exports[`TestComponent: 0 1`] = `
<div>
  hello world
</div>
`;
